pkg/pki/pkcs7/pkcs7.go

```diff
@@ -213,3 +215,27 @@ func (k PublicKey) EmailAddresses() []string {
 func (k PublicKey) Subjects() []string {
 	return k.EmailAddresses()
 }
+
+// Identities implements the pki.PublicKey interface
+func (k PublicKey) Identities() ([]string, error) {
+	// pkcs7 structure may contain both a key and certificate chain
+	var identities []string
+	if k.key != nil {
+		pem, err := cryptoutils.MarshalPublicKeyToPEM(k.key)
+		if err != nil {
+			return nil, err
+		}
+		identities = append(identities, string(pem))
+
+	}
+	if len(k.certs) > 0 {
+		cert := k.certs[0]
+		pem, err := cryptoutils.MarshalPublicKeyToPEM(cert.PublicKey)
+		if err != nil {
+			return nil, err
+		}
+		identities = append(identities, string(pem))
+		identities = append(identities, px509.GetSubjectAlternateNames(cert)...)
```

Comment:
Changed for Subjects(), and added comment.

Reply:
These actually aren't equivalent currently, it looks like EmailAddresses is searching the cert's subject rather than SAN. Do you know what's expected for a pkcs7 certificate? The example test cert includes an email in the subject rather than SAN.

Reply:
To be safe, maybe search the SAN and subject?

Reply:
I search both now.